### PR TITLE
CLOUDP-94119: Do not mark the CR as failed on secret absence, it could be deleted at later stage after when scram is set

### PR DESCRIPTION
**Pull request by irajdeep**
### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you signed all of your commits?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: controllers/mongodb_users.go [hunk 1]
```diff
@@ -5,18 +5,26 @@ import (
...
 // and will start monitor them so that the reconcile process is triggered every time these secrets are updated
-func (r ReplicaSetReconciler) ensureUserResources(mdb mdbv1.MongoDBCommunity) error {
+func (r ReplicaSetReconciler) ensureUserResources(mdb mdbv1.MongoDBCommunity, log *zap.SugaredLogger) error {
 	for _, user := range mdb.GetScramUsers() {
 		secretNamespacedName := types.NamespacedName{Name: user.PasswordSecretName, Namespace: mdb.Namespace}
 		if _, err := secret.ReadKey(r.client, user.PasswordSecretKey, secretNamespacedName); err != nil {
 			if apiErrors.IsNotFound(err) {
-				return fmt.Errorf(`user password secret "%s" not found: %s`, secretNamespacedName, err)
+				// check for SCRAM secret as well
+				scramSecretName := mdb.GetAgentScramCredentialsNamespacedName()
```
Comment (chatton) [inline controllers/mongodb_users.go:21]: this should be `user.GetScramCredentialsSecretName()` (there is a different set of creds for the Agent)

File: controllers/mongodb_users.go [hunk 2]
```diff
@@ -5,18 +5,26 @@ import (
...
 	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/secret"
+	"go.uber.org/zap"
 	apiErrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 )
 
 // ensureUserResources will check that the configured user password secrets can be found
 // and will start monitor them so that the reconcile process is triggered every time these secrets are updated
-func (r ReplicaSetReconciler) ensureUserResources(mdb mdbv1.MongoDBCommunity) error {
+func (r ReplicaSetReconciler) ensureUserResources(mdb mdbv1.MongoDBCommunity, log *zap.SugaredLogger) error {
```
Comment (chatton) [inline controllers/mongodb_users.go:15]: no need to `log` argument (it's on the type already)

**Review by chatton — COMMENTED** [review]
Looks good once we correct the secret name 👍

File: controllers/mongodb_users.go (hunk 1 above)
Comment (irajdeep) [reply controllers/mongodb_users.go:21]: there seems to be leftover `GetAgentScramCredentialsNamespacedName` which is not used anymore and we have just one credentials -- so I will remove the former.

**Pull request merged**